test/jdk/javax/swing/JScrollPane/AcceleratedWheelScrolling/RTLScrollers.java

```diff
@@ -0,0 +1,677 @@
+        tableScroller.addMouseWheelListener(this);
+
+        GridBagConstraints tableGBC = new GridBagConstraints(0,
+                0,
+                1,
+                1,
+                1.0,
+                0.3,
+                GridBagConstraints.CENTER,
+                GridBagConstraints.BOTH,
+                new Insets(0,0,0,0),
+                0,
+                0);
+        content.add(tableScroller, tableGBC);
+        GridBagConstraints textGBC = new GridBagConstraints(0,
+                1,
+                1,
+                1,
+                1.0,
+                0.3,
+                GridBagConstraints.CENTER,
+                GridBagConstraints.BOTH,
+                new Insets(0,0,0,0),
+                0,
+                0);
+        content.add(textScroller, textGBC);
+
+        GridBagConstraints listGBC = new GridBagConstraints(0,
+                2,
+                1,
+                5,
+                1.0,
+                1.2,
+                GridBagConstraints.CENTER,
+                GridBagConstraints.BOTH,
+                new Insets(0,0,0,0),
+                0,
+                0);
+
+        content.add(listScroller, listGBC);
+
+        rightToLeft = new JCheckBoxMenuItem("Right-To-Left", false);
+        rightToLeft.addActionListener(this);
+        JMenu menu = new JMenu("Component Orientation");
+        menu.add(rightToLeft);
+
+        JMenuItem close = new JMenuItem("Close");
+        close.addActionListener(e -> RTLScrollers.this.setVisible(false));
+        menu.add(close);
+
+        JMenuBar mb = new JMenuBar();
+        mb.add(menu);
+        setJMenuBar(mb);
+        setBounds(0, 0, WINWIDTH, 760);
+        setLocationRelativeTo(null);
+    }
+
+    public void actionPerformed(ActionEvent e) {
+        if (rightToLeft.getState()) {
+            applyComponentOrientation(ComponentOrientation.RIGHT_TO_LEFT);
+        }
+        else {
+            applyComponentOrientation(ComponentOrientation.LEFT_TO_RIGHT);
+        }
+    }
+    public void mouseWheelMoved(MouseWheelEvent e) {
+        System.out.println("Rotation: " + e.getWheelRotation());
+    }
+
+    public static boolean runTest(int scrollAmount)
+            throws InterruptedException, InvocationTargetException {
+        System.out.println("RTLS.runTest()");
+        if (robot == null) {
+            try {
+                robot = new Robot();
+                robot.setAutoDelay(150);
+                robot.setAutoWaitForIdle(true);
+            }
+            catch (AWTException e) {
+                e.printStackTrace();
+                return false;
+            }
+        }
+
+        SwingUtilities.invokeAndWait(() -> {
+            rtl = new RTLScrollers(scrollAmount);
+            rtl.setVisible(true);
+        });
+        robot.delay(100);
+
+        SwingUtilities.invokeAndWait(() -> {
+            try {
+                retVal = rtl.runTests(scrollAmount);
+                rtl.setVisible(false);
```

Comment:
Since you are creating an instance for each test scenario i would put the setVisible(false) into the finally {} clause so it will be called even if runTests throws an exception and add the dispose() call for good measure.